# 📊 EDA AVANZADO - Predicción de Deserción Estudiantil
## Dataset: UCI Repository - Students Dropout and Academic Success
---
**Objetivo:** Análisis exploratorio profundo para identificar patrones, correlaciones y features clave.

**Duración estimada:** 3 horas

**Autor:** Jhordan Cotrina  
**Fecha:** Octubre 2025

## 📦 1. IMPORTACIONES Y CONFIGURACIÓN

In [1]:
# Librerías estándar
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px
import plotly.graph_objects as go
from plotly.subplots import make_subplots
import warnings

import IPython.display as display, html

# Configuración
warnings.filterwarnings('ignore')
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', 100)
plt.style.use('seaborn-v0_8-darkgrid')
sns.set_palette("husl")

# Configuración de Plotly
import plotly.io as pio
pio.renderers.default = "notebook_connected"

print("✅ Librerías importadas correctamente")

✅ Librerías importadas correctamente


## 📥 2. CARGA DE DATOS

In [ ]:
# Cargar dataset
df = pd.read_csv('../data/raw/student_data.csv')

print(f"📊 Dataset cargado: {df.shape[0]} filas × {df.shape[1]} columnas")
print(f"\n📋 Columnas:")
for i, col in enumerate(df.columns, 1):
    print(f"  {i:2d}. {col}")

📊 Dataset cargado: 4424 filas × 37 columnas

📋 Columnas:
   1. Marital status
   2. Application mode
   3. Application order
   4. Course
   5. Daytime/evening attendance	
   6. Previous qualification
   7. Previous qualification (grade)
   8. Nacionality
   9. Mother's qualification
  10. Father's qualification
  11. Mother's occupation
  12. Father's occupation
  13. Admission grade
  14. Displaced
  15. Educational special needs
  16. Debtor
  17. Tuition fees up to date
  18. Gender
  19. Scholarship holder
  20. Age at enrollment
  21. International
  22. Curricular units 1st sem (credited)
  23. Curricular units 1st sem (enrolled)
  24. Curricular units 1st sem (evaluations)
  25. Curricular units 1st sem (approved)
  26. Curricular units 1st sem (grade)
  27. Curricular units 1st sem (without evaluations)
  28. Curricular units 2nd sem (credited)
  29. Curricular units 2nd sem (enrolled)
  30. Curricular units 2nd sem (evaluations)
  31. Curricular units 2nd sem (approved)
  32.

## 🔍 3. ANÁLISIS ESTADÍSTICO DESCRIPTIVO

In [3]:
# Información general
print("="*60)
print("INFORMACIÓN GENERAL DEL DATASET")
print("="*60)
df.info()

INFORMACIÓN GENERAL DEL DATASET
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4424 entries, 0 to 4423
Data columns (total 37 columns):
 #   Column                                          Non-Null Count  Dtype  
---  ------                                          --------------  -----  
 0   Marital status                                  4424 non-null   int64  
 1   Application mode                                4424 non-null   int64  
 2   Application order                               4424 non-null   int64  
 3   Course                                          4424 non-null   int64  
 4   Daytime/evening attendance	                     4424 non-null   int64  
 5   Previous qualification                          4424 non-null   int64  
 6   Previous qualification (grade)                  4424 non-null   float64
 7   Nacionality                                     4424 non-null   int64  
 8   Mother's qualification                          4424 non-null   int64  
 9   Father's 

In [4]:
# Estadísticas descriptivas
print("\n" + "="*60)
print("ESTADÍSTICAS DESCRIPTIVAS")
print("="*60)
df.describe().round(2)


ESTADÍSTICAS DESCRIPTIVAS


,Marital status,Application mode,Application order,Course,Daytime/evening attendance\t,Previous qualification,Previous qualification (grade),Nacionality,Mother's qualification,Father's qualification,Mother's occupation,Father's occupation,Admission grade,Displaced,Educational special needs,Debtor,Tuition fees up to date,Gender,Scholarship holder,Age at enrollment,International,Curricular units 1st sem (credited),Curricular units 1st sem (enrolled),Curricular units 1st sem (evaluations),Curricular units 1st sem (approved),Curricular units 1st sem (grade),Curricular units 1st sem (without evaluations),Curricular units 2nd sem (credited),Curricular units 2nd sem (enrolled),Curricular units 2nd sem (evaluations),Curricular units 2nd sem (approved),Curricular units 2nd sem (grade),Curricular units 2nd sem (without evaluations),Unemployment rate,Inflation rate,GDP
count,4424.00,4424.00,4424.00,4424.00,4424.00,4424.00,4424.00,4424.00,4424.00,4424.00,4424.00,4424.00,4424.00,4424.00,4424.00,4424.00,4424.00,4424.00,4424.00,4424.00,4424.00,4424.00,4424.00,4424.00,4424.00,4424.00,4424.00,4424.00,4424.00,4424.00,4424.00,4424.00,4424.00,4424.00,4424.00,4424.00
mean,1.18,18.67,1.73,8856.64,0.89,4.58,132.61,1.87,19.56,22.28,10.96,11.03,126.98,0.55,0.01,0.11,0.88,0.35,0.25,23.27,0.02,0.71,6.27,8.30,4.71,10.64,0.14,0.54,6.23,8.06,4.44,10.23,0.15,11.57,1.23,0.00
std,0.61,17.48,1.31,2063.57,0.31,10.22,13.19,6.91,15.60,15.34,26.42,25.26,14.48,0.50,0.11,0.32,0.32,0.48,0.43,7.59,0.16,2.36,2.48,4.18,3.09,4.84,0.69,1.92,2.20,3.95,3.01,5.21,0.75,2.66,1.38,2.27
min,1.00,1.00,0.00,33.00,0.00,1.00,95.00,1.00,1.00,1.00,0.00,0.00,95.00,0.00,0.00,0.00,0.00,0.00,0.00,17.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,7.60,-0.80,-4.06
25%,1.00,1.00,1.00,9085.00,1.00,1.00,125.00,1.00,2.00,3.00,4.00,4.00,117.90,0.00,0.00,0.00,1.00,0.00,0.00,19.00,0.00,0.00,5.00,6.00,3.00,11.00,0.00,0.00,5.00,6.00,2.00,10.75,0.00,9.40,0.30,-1.70
50%,1.00,17.00,1.00,9238.00,1.00,1.00,133.10,1.00,19.00,19.00,5.00,7.00,126.10,1.00,0.00,0.00,1.00,0.00,0.00,20.00,0.00,0.00,6.00,8.00,5.00,12.29,0.00,0.00,6.00,8.00,5.00,12.20,0.00,11.10,1.40,0.32
75%,1.00,39.00,2.00,9556.00,1.00,1.00,140.00,1.00,37.00,37.00,9.00,9.00,134.80,1.00,0.00,0.00,1.00,1.00,0.00,25.00,0.00,0.00,7.00,10.00,6.00,13.40,0.00,0.00,7.00,10.00,6.00,13.33,0.00,13.90,2.60,1.79
max,6.00,57.00,9.00,9991.00,1.00,43.00,190.00,109.00,44.00,44.00,194.00,195.00,190.00,1.00,1.00,1.00,1.00,1.00,1.00,70.00,1.00,20.00,26.00,45.00,26.00,18.88,12.00,19.00,23.00,33.00,20.00,18.57,12.00,16.20,3.70,3.51


## 🎯 4. ANÁLISIS DEL TARGET (Variable Objetivo)

In [8]:
# Identificar columna target
target_col = 'Target'  # Ajustar si el nombre es diferente

if target_col in df.columns:
    print(f"✅ Target encontrado: {target_col}")
    print(f"\n📊 Distribución de clases:")
    print(df[target_col].value_counts())
    print(f"\n📈 Porcentajes:")
    print(df[target_col].value_counts(normalize=True).mul(100).round(2))
else:
    print("⚠️ No se encontró la columna 'Target'. Columnas disponibles:")
    print(df.columns.tolist())

✅ Target encontrado: Target

📊 Distribución de clases:
Target
Graduate    2209
Dropout     1421
Enrolled     794
Name: count, dtype: int64

📈 Porcentajes:
Target
Graduate    49.93
Dropout     32.12
Enrolled    17.95
Name: proportion, dtype: float64


In [9]:
# Visualización interactiva del target
if target_col in df.columns:
    fig = px.pie(
        df[target_col].value_counts().reset_index(),
        values='count',
        names=target_col,
        title='📊 Distribución de Clases (Target)',
        hole=0.4,
        color_discrete_sequence=px.colors.qualitative.Set3
    )
    fig.update_traces(textposition='inside', textinfo='percent+label')
    fig.update_layout(height=500)
    
    fig.show()
    
    # Análisis de desbalanceo
    class_counts = df[target_col].value_counts()
    majority_class = class_counts.max()
    minority_class = class_counts.min()
    imbalance_ratio = majority_class / minority_class
    
    print(f"\n⚖️ ANÁLISIS DE DESBALANCEO:")
    print(f"  • Clase mayoritaria: {class_counts.idxmax()} ({majority_class} muestras)")
    print(f"  • Clase minoritaria: {class_counts.idxmin()} ({minority_class} muestras)")
    print(f"  • Ratio de desbalanceo: {imbalance_ratio:.2f}:1")
    
    if imbalance_ratio > 3:
        print(f"  ⚠️ DATASET DESBALANCEADO - Considerar SMOTE o class_weight")
    else:
        print(f"  ✅ Dataset relativamente balanceado")


⚖️ ANÁLISIS DE DESBALANCEO:
  • Clase mayoritaria: Graduate (2209 muestras)
  • Clase minoritaria: Enrolled (794 muestras)
  • Ratio de desbalanceo: 2.78:1
  ✅ Dataset relativamente balanceado


## 🔎 5. DETECCIÓN DE VALORES FALTANTES

In [10]:
# Análisis de missing values
missing = df.isnull().sum()
missing_pct = (missing / len(df) * 100).round(2)
missing_df = pd.DataFrame({
    'Missing_Count': missing,
    'Missing_Percentage': missing_pct
}).sort_values('Missing_Count', ascending=False)

missing_df = missing_df[missing_df['Missing_Count'] > 0]

if len(missing_df) > 0:
    print("⚠️ COLUMNAS CON VALORES FALTANTES:")
    print(missing_df)
    
    # Visualización
    fig = px.bar(
        missing_df.reset_index(),
        x='index',
        y='Missing_Percentage',
        title='📉 Porcentaje de Valores Faltantes por Columna',
        labels={'index': 'Columna', 'Missing_Percentage': '% Faltante'}
    )
    fig.update_layout(xaxis_tickangle=-45, height=500)
    fig.show()
else:
    print("✅ NO HAY VALORES FALTANTES - Dataset completo")

✅ NO HAY VALORES FALTANTES - Dataset completo


## 📊 6. ANÁLISIS DE FEATURES NUMÉRICAS

In [11]:
# Identificar columnas numéricas
numeric_cols = df.select_dtypes(include=[np.number]).columns.tolist()
print(f"📊 Features numéricas: {len(numeric_cols)}")
print(numeric_cols[:10], "..." if len(numeric_cols) > 10 else "")

📊 Features numéricas: 36
['Marital status', 'Application mode', 'Application order', 'Course', 'Daytime/evening attendance\t', 'Previous qualification', 'Previous qualification (grade)', 'Nacionality', "Mother's qualification", "Father's qualification"] ...


In [12]:
# Distribuciones de features numéricas clave (primeras 6)
key_numeric_features = numeric_cols[:6]

fig = make_subplots(
    rows=2, cols=3,
    subplot_titles=key_numeric_features,
    vertical_spacing=0.12,
    horizontal_spacing=0.1
)

for i, col in enumerate(key_numeric_features, 1):
    row = (i - 1) // 3 + 1
    col_pos = (i - 1) % 3 + 1
    
    fig.add_trace(
        go.Histogram(x=df[col], name=col, showlegend=False),
        row=row, col=col_pos
    )

fig.update_layout(height=600, title_text="📊 Distribuciones de Features Numéricas Clave")
fig.show()

## 🎨 7. ANÁLISIS DE CORRELACIONES

In [13]:
# Matriz de correlación
correlation_matrix = df[numeric_cols].corr()

# Heatmap interactivo
fig = px.imshow(
    correlation_matrix,
    text_auto='.2f',
    aspect='auto',
    color_continuous_scale='RdBu_r',
    color_continuous_midpoint=0,
    title='🔥 Matriz de Correlación - Features Numéricas'
)
fig.update_layout(height=800, width=900)
fig.show()

In [14]:
# Análisis de features numéricas vs target categórico
from scipy.stats import f_oneway

print("=" * 60)
print("ANÁLISIS: Features Numéricas vs Target Categórico")
print("=" * 60)

# Calcular ANOVA F-statistic para cada feature numérica
anova_results = []

for col in numeric_cols:
    if col != target_col:  # Excluir el target si estuviera como numérico
        groups = [df[df[target_col] == category][col].dropna() 
                  for category in df[target_col].unique()]
        
        # Solo calcular si hay suficientes datos
        if all(len(g) > 0 for g in groups):
            f_stat, p_value = f_oneway(*groups)
            anova_results.append({
                'Feature': col,
                'F_Statistic': f_stat,
                'P_Value': p_value,
                'Significativo': 'Sí' if p_value < 0.05 else 'No'
            })

# Crear DataFrame y ordenar por F-statistic
anova_df = pd.DataFrame(anova_results).sort_values('F_Statistic', ascending=False)

print("\nTop 15 Features más relacionadas con el Target (por ANOVA):")
print(anova_df.head(15).to_string(index=False))

# Visualización
top_15 = anova_df.head(15)
fig = px.bar(
    top_15,
    x='F_Statistic',
    y='Feature',
    orientation='h',
    title='Top 15 Features por Importancia (ANOVA F-Statistic)',
    labels={'F_Statistic': 'F-Statistic', 'Feature': 'Feature'},
    color='F_Statistic',
    color_continuous_scale='Viridis'
)
fig.update_layout(height=600, yaxis={'categoryorder':'total ascending'})
fig.show()

print(f"\nInterpretación:")
print(f"- F-Statistic alto = La feature separa bien las clases del target")
print(f"- P-Value < 0.05 = Relación estadísticamente significativa")
print(f"\nFeatures significativas: {anova_df[anova_df['P_Value'] < 0.05].shape[0]} de {len(anova_df)}")

ANÁLISIS: Features Numéricas vs Target Categórico

Top 15 Features más relacionadas con el Target (por ANOVA):
                               Feature  F_Statistic       P_Value Significativo
   Curricular units 2nd sem (approved)  1410.732938  0.000000e+00            Sí
      Curricular units 2nd sem (grade)  1134.109544  0.000000e+00            Sí
   Curricular units 1st sem (approved)   859.866768 3.649472e-316            Sí
      Curricular units 1st sem (grade)   713.517328 2.803052e-269            Sí
               Tuition fees up to date   505.621429 1.784950e-198            Sí
                    Scholarship holder   225.751437  4.436825e-94            Sí
                     Age at enrollment   154.712071  1.138849e-65            Sí
                                Debtor   137.647527  1.018223e-58            Sí
                                Gender   123.041811  9.950346e-53            Sí
                      Application mode   114.534956  3.190893e-49            Sí
Curricula


Interpretación:
- F-Statistic alto = La feature separa bien las clases del target
- P-Value < 0.05 = Relación estadísticamente significativa

Features significativas: 31 de 36


## 📦 8. DETECCIÓN DE OUTLIERS

In [15]:
# Boxplots para detectar outliers (primeros 6 features)
key_features_for_outliers = numeric_cols[:6]

fig = make_subplots(
    rows=2, cols=3,
    subplot_titles=key_features_for_outliers,
    vertical_spacing=0.15,
    horizontal_spacing=0.1
)

for i, col in enumerate(key_features_for_outliers, 1):
    row = (i - 1) // 3 + 1
    col_pos = (i - 1) % 3 + 1
    
    fig.add_trace(
        go.Box(y=df[col], name=col, showlegend=False),
        row=row, col=col_pos
    )

fig.update_layout(height=600, title_text="📦 Boxplots - Detección de Outliers")
fig.show()

In [21]:
# Separar features numéricas continuas de categóricas codificadas
continuous_features = [
    'Age at enrollment',
    'Previous qualification (grade)',
    'Admission grade',
    'Curricular units 1st sem (credited)',
    'Curricular units 1st sem (enrolled)',
    'Curricular units 1st sem (evaluations)',
    'Curricular units 1st sem (approved)',
    'Curricular units 1st sem (grade)',
    'Curricular units 2nd sem (credited)',
    'Curricular units 2nd sem (enrolled)',
    'Curricular units 2nd sem (evaluations)',
    'Curricular units 2nd sem (approved)',
    'Curricular units 2nd sem (grade)',
    'Unemployment rate',
    'Inflation rate',
    'GDP'
]

# Filtrar solo las que existen en tu dataset
continuous_features = [col for col in continuous_features if col in df.columns]

print(f"Features numéricas continuas: {len(continuous_features)}")
print(continuous_features)

Features numéricas continuas: 16
['Age at enrollment', 'Previous qualification (grade)', 'Admission grade', 'Curricular units 1st sem (credited)', 'Curricular units 1st sem (enrolled)', 'Curricular units 1st sem (evaluations)', 'Curricular units 1st sem (approved)', 'Curricular units 1st sem (grade)', 'Curricular units 2nd sem (credited)', 'Curricular units 2nd sem (enrolled)', 'Curricular units 2nd sem (evaluations)', 'Curricular units 2nd sem (approved)', 'Curricular units 2nd sem (grade)', 'Unemployment rate', 'Inflation rate', 'GDP']


In [22]:
# Análisis de outliers SOLO en variables continuas
print("ANÁLISIS DE OUTLIERS (Solo variables continuas):")
print("="*60)
outlier_summary = []

for col in continuous_features:
    Q1 = df[col].quantile(0.25)
    Q3 = df[col].quantile(0.75)
    IQR = Q3 - Q1
    lower_bound = Q1 - 1.5 * IQR
    upper_bound = Q3 + 1.5 * IQR
    outliers = df[(df[col] < lower_bound) | (df[col] > upper_bound)]
    count = len(outliers)
    pct = (count / len(df) * 100)
    
    outlier_summary.append({
        'Feature': col,
        'Outliers_Count': count,
        'Outliers_Pct': round(pct, 2)
    })
    
    if pct > 5:
        print(f"⚠️ {col}: {count} outliers ({pct:.2f}%)")

outlier_df = pd.DataFrame(outlier_summary).sort_values('Outliers_Pct', ascending=False)
print(f"\nResumen:")
print(outlier_df)

ANÁLISIS DE OUTLIERS (Solo variables continuas):
⚠️ Age at enrollment: 441 outliers (9.97%)
⚠️ Curricular units 1st sem (credited): 577 outliers (13.04%)
⚠️ Curricular units 1st sem (enrolled): 424 outliers (9.58%)
⚠️ Curricular units 1st sem (grade): 726 outliers (16.41%)
⚠️ Curricular units 2nd sem (credited): 530 outliers (11.98%)
⚠️ Curricular units 2nd sem (enrolled): 369 outliers (8.34%)
⚠️ Curricular units 2nd sem (grade): 877 outliers (19.82%)

Resumen:
                                   Feature  Outliers_Count  Outliers_Pct
12        Curricular units 2nd sem (grade)             877         19.82
7         Curricular units 1st sem (grade)             726         16.41
3      Curricular units 1st sem (credited)             577         13.04
8      Curricular units 2nd sem (credited)             530         11.98
0                        Age at enrollment             441          9.97
4      Curricular units 1st sem (enrolled)             424          9.58
9      Curricular units

In [24]:
# Buscar valores imposibles en calificaciones (deben estar 0-20)
print("Verificación de rangos válidos:")
print(f"Grades 1st sem: {df['Curricular units 1st sem (grade)'].min()} - {df['Curricular units 1st sem (grade)'].max()}")
print(f"Grades 2nd sem: {df['Curricular units 2nd sem (grade)'].min()} - {df['Curricular units 2nd sem (grade)'].max()}")
print(f"Age: {df['Age at enrollment'].min()} - {df['Age at enrollment'].max()}")

Verificación de rangos válidos:
Grades 1st sem: 0.0 - 18.875
Grades 2nd sem: 0.0 - 18.571428571428573
Age: 17 - 70


## 🎯 9. ANÁLISIS BIVARIADO: TARGET vs FEATURES

In [26]:
# Visualizar las features que SÍ separan las clases
important_features = [
    'Curricular units 1st sem (approved)',
    'Curricular units 1st sem (grade)',
    'Curricular units 2nd sem (approved)',
    'Curricular units 2nd sem (grade)'
]

fig = make_subplots(
    rows=2, cols=2,
    subplot_titles=important_features,
    vertical_spacing=0.15,
    horizontal_spacing=0.1
)

for i, col in enumerate(important_features, 1):
    row = (i - 1) // 2 + 1
    col_pos = (i - 1) % 2 + 1
    
    for target_value in df[target_col].unique():
        data = df[df[target_col] == target_value][col]
        fig.add_trace(
            go.Box(y=data, name=str(target_value), showlegend=(i==1)),
            row=row, col=col_pos
        )

fig.update_layout(height=700, title_text="Variables Académicas por Clase (Target)")
fig.show()

## 📋 10. INSIGHTS Y CONCLUSIONES

### 📝 Resumen del Análisis:

**1. Calidad de los Datos:**
- ✅ Valores faltantes: Ninguno - Dataset 100% completo (4424 × 36)
- ⚠️ Outliers: 20% en calificaciones, 10% en edad

-  Decisión: MANTENER todos los outliers (son casos legítimos de deserción)
-  Modelos tree-based (CatBoost/LightGBM) son robustos a outliers

- ✅ Distribución del target:  

- Graduate: 50%, Dropout: 32%, Enrolled: 18%
- Ratio 2.78:1 (manejable sin SMOTE)

**2. Features Más Relevantes:**

Curricular units 2nd sem (approved) - F=1410 ⭐
Curricular units 2nd sem (grade) - F=1134 ⭐
Curricular units 1st sem (approved) - F=860 ⭐
Curricular units 1st sem (grade) - F=714 ⭐
Tuition fees up to date - F=506 (factor económico)

Insight crítico: Las 4 variables académicas del 1er y 2do semestre son los predictores más fuertes.

**3. Desbalanceo de Clases:**

- Ratio: 2.78:1 (Graduate vs Enrolled)
- Estrategia: class_weight='balanced' (NO necesitas SMOTE inicialmente)
- Métrica prioritaria: Recall en clase "Dropout" (tu objetivo A+C)

**4. Estrategia de Preprocessing:**
- [✅] Imputación: NO necesaria (0 missing values)
- [✅] Outliers: MANTENER (información valiosa)
- [ ] Feature scaling: StandardScaler para variables continuas
- [ ] Feature engineering: Crear "semestre_promedio" = (1st_grade + 2nd_grade) / 2
                           Crear "tasa_aprobacion" = approved / enrolled
                           Binario "riesgo_temprano" = 1 si <4 aprobadas en 1er sem
- [ ]  Encoding: One-Hot para categóricas (Course, Marital status, etc.)
- [✅] Desbalanceo: class_weight='balanced' en modelos

**5.Reglas de Deteccion**

- 🚨 Riesgo ALTO (1er semestre):

Aprobadas < 4 unidades Y promedio < 11


- ⚠️ Confirmación (2do semestre):

Si patrón continúa → Intervención urgente

**6. Próximos Pasos:**

- Día 3: Feature engineering + Pipeline de preprocessing
- Día 4-5: Entrenamiento modelos (CatBoost, LightGBM, XGBoost)
- Día 6: Tuning con Optuna (optimizar recall en Dropout)
- Día 7: Registrar en MLflow + Deployment

## 💾 11. GUARDAR RESULTADOS

In [27]:
# Ejecuta esto en tu notebook (Sección 11)
import json

insights = {
    'dataset_shape': [4424, 36],
    'target_distribution': {'Graduate': 2209, 'Dropout': 1421, 'Enrolled': 794},
    'missing_values': 0,
    'outliers_pct': 20,
    'top_5_features': [
        'Curricular units 2nd sem (approved)',
        'Curricular units 2nd sem (grade)',
        'Curricular units 1st sem (approved)',
        'Curricular units 1st sem (grade)',
        'Tuition fees up to date'
    ],
    'early_detection_rule': 'approved < 4 AND grade < 11 in 1st semester',
    'preprocessing_strategy': 'StandardScaler + class_weight=balanced + maintain outliers',
    'next_step': 'Feature Engineering'
}

# Crear carpeta si no existe
import os
os.makedirs('../reports', exist_ok=True)

with open('../../reports/eda_insights.json', 'w') as f:
    json.dump(insights, f, indent=2)

print("✅ Insights guardados en reports/eda_insights.json")

✅ Insights guardados en reports/eda_insights.json


In [ ]:
# Guardar versión limpia del dataset (si hiciste cambios)
# df.to_csv('../data/processed/student_data_clean.csv', index=False)
# print("✅ Dataset procesado guardado")

---
## ✅ FIN DEL ANÁLISIS EDA

**Siguiente paso:** Feature Engineering y Pipeline de Preprocessing  
**Notebook:** `03_Feature_Engineering.ipynb`